[Ong Weng Qi] iP

chriswengqi opened the pull request:
[](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html#dukepro)DukePro
“Never put off till tomorrow what may be done day after tomorrow just as well.” – Mark Twain [(source)](http://www.twainquotes.com/Procrastination.html)

DukePro is an ultimate procrastination killer which helps you remember what to do so that you cannot use this as an excuse to skip your tasks! It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

chriswengqi committed:
Level-1

chriswengqi committed:
Level-2

chriswengqi committed:
Level-3

chriswengqi committed:
Level-4

chriswengqi committed:
Level-5

chriswengqi committed:
Level-6

chriswengqi committed:
Level-6 corrected

chriswengqi committed:
A-TextUiTesting

chriswengqi committed:
Complete Level 7

chriswengqi committed:
Completed Level 8

chriswengqi committed:
Merge branch 'branch-Level-7'

chriswengqi committed:
Merge branch 'branch-Level-8'

chriswengqi committed:
Add Ui class

chriswengqi committed:
Add Storage class

chriswengqi committed:
Add Parser class

chriswengqi committed:
Add TaskList and minor bug fixes

chriswengqi committed:
Add more classes and minor bug fixes

chriswengqi committed:
Create packages

chriswengqi committed:
Create packages

chriswengqi committed:
Add J-Unit testcases

chriswengqi committed:
Create JAR file and make more OOP changes

chriswengqi committed:
Add JavaDocs comments

chriswengqi committed:
Fix coding standards

chriswengqi committed:
Complete Level-9

chriswengqi committed:
Merge branch 'branch-A-JavaDoc'

chriswengqi committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke.java

chriswengqi committed:
Merge branch 'branch-Level-9'

chriswengqi committed:
Merge pull request #3 from chriswengqi/branch-A-JavaDoc

Add JavaDocs comments

chriswengqi committed:
Merge pull request #2 from chriswengqi/branch-Level-9

Complete Level-9

chriswengqi committed:
Merge branch 'master' into branch-A-CodingStandard

chriswengqi committed:
Merge pull request #1 from chriswengqi/branch-A-CodingStandard

Fix coding standards

chriswengqi committed:
Merge branch 'master' of https://github.com/chriswengqi/ip

# Conflicts:
#	src/main/java/Duke.java

brigittesantoso commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,152 @@
+
+    }
+
+    public void run() {
```
Avoid long methods. Try to split this method up into smaller parts.
![image](https://user-images.githubusercontent.com/80502407/151958785-f21590ed-a80b-470a-897a-0e2239eb2e3f.png)

SethCKL replied:
agreed, can potentially abstract out certain parts of the code to another class or to another function

brigittesantoso commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,152 @@
+        String item = parser.getItem();
+
+
+        while (!task.equals("bye")) {
```
Each string here can be split into separate methods to improve readability of the code. 

Example:
public void deadlineTask() {
         if (!item.equals("")) {
                    String thing = item.split(" /by ")[0];
                    String time = item.split(" /by ")[1];
                    tasks.add(new Deadline(thing, time));
         } else {
                    ui.reply("Can read instructions or not? Deadline cannot be empty :/");
         }
}

Thus inside this while loop it is just:
while ...
            } else if (task.equals("deadline")) {
                          deadlineTask();
           }
...

brigittesantoso commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,33 @@
+public class Event extends Task {
+
+    protected LocalDateTime ldt;
+    public String at;
```
Class variables should not be public.
![image](https://user-images.githubusercontent.com/80502407/151959418-4756526e-dfb5-49d8-abfb-1670c316fcb5.png)

brigittesantoso commented on src/main/java/task/TaskListTest.java:
```diff
@@ -0,0 +1,26 @@
+        TaskList tl = new TaskList();
+        assertEquals("Ah sure. I've added this task:\n" +
+                "      [T][ ] sleep\n" +
+                "    Now you have 1 task in the list.", tl.addTask(new Todo("sleep"), 1));
```
Spacing on this line should be 12 instead of 8 as it is a wrapped line. Same for below case under method removeTask()

brigittesantoso commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,84 @@
+    private static Task parseTextFile(String s) {
+        Task t;
+        String[] strarr = s.split(" ");
+        if (s.contains("[X]")) {
```
Code readability can be improved for this if-else part.

brigittesantoso commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,84 @@
+
+public class Storage {
+
+    File file;
```
Class variable scope should be declared and should be non-public.

brigittesantoso commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+import java.util.List;
+
+public class TaskList {
+    public List<Task> list;
```
Class variables should never be declared as public.
![image](https://user-images.githubusercontent.com/80502407/151960241-82ced349-a4d9-4306-8d9c-7a7229d4306d.png)

brigittesantoso commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,84 @@
+            }
+        }
+
+    private static Task parseTextFile(String s) {
```
Naming of variables here can be improved instead of "String s" and "Task t"

brigittesantoso reviewed (commented):
Good job! Just a few minor changes needed to improve code quality 👍

chriswengqi committed:
Merge remote-tracking branch 'origin/add-gradle-support'

chriswengqi committed:
Fix checkstyle errors

SethCKL commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,35 @@
+
+public class Event extends Task {
+
+    public String at;
```
Agree with brigittesantoso, recommend setting access modifier to private

SethCKL commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,125 @@
+import java.util.List;
+
+public class TaskList {
+    public List<Task> list;
```
same as above and in agreement with brigittesantoso, recommend setting variables access modifier to private always, unless the usecase or specific situation calls for otherwise

SethCKL commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+                } else {
+                    ui.reply("Can read instructions or not? Event cannot be empty :/");
+                }
+
```
would suggest grouping up the switch statements to do away the duplicated code.

case "todo":
case "deadline":
case "event":

if (!item.equals("")) {
Check if it's deadline/event. if so call the additional split functions.
else, just create a todo task
then add task at the end

}

chriswengqi committed:
Add Level 10

chriswengqi committed:
Fix checkstyle and bugs

chriswengqi committed:
There is a lack of assetion statements

Assertion statements are essential to detect potential bugs

Add more assertion statements

To check whether exceptions are handled correctly during runtime.

chriswengqi committed:
Code consist of methods that are too long and not abstracted well

Change to improve code quality

Change methods that are too long and change method names for easier readability of user

For better ccode readability

chriswengqi committed:
Add extension to suuport managing contacts

chriswengqi committed:
Add contact commands to CommandList

chriswengqi committed:
Fix code quality

chriswengqi committed:
Merge branch 'branch-A-CodeQuality'

chriswengqi committed:
add assertions to ContactList for managing contacts

chriswengqi committed:
Merge branch 'branch-A-Assertions'

chriswengqi committed:
Update User Guide

chriswengqi committed:
Merge branch 'branch-A-Assertions'

chriswengqi committed:
Update README

chriswengqi committed:
Fix issues on IOException when text file is not found

chriswengqi committed:
Fix bug where marked Todo task becomes unmarked when restarting Snorlax
